pkg/kclient/deployments.go

```diff
@@ -185,18 +190,40 @@ See below for a list of failed events that occured more than %d times during dep
 	}
 }
 
-// CreateDeployment creates a deployment based on the given deployment spec
+func resourceAsJson(resource interface{}) string {
+	data, _ := json.MarshalIndent(resource, " ", " ")
+	return string(data)
+}
+
+// ApplyDeployment updates a deployment based on the given deployment spec
 func (c *Client) CreateDeployment(deploy appsv1.Deployment) (*appsv1.Deployment, error) {
-	deployment, err := c.KubeClient.AppsV1().Deployments(c.Namespace).Create(&deploy)
+	deployment, err := c.KubeClient.AppsV1().Deployments(c.Namespace).Create(context.TODO(), &deploy, metav1.CreateOptions{FieldManager: "odo"})
 	if err != nil {
-		return nil, errors.Wrapf(err, "unable to create Deployment %s", deploy.Name)
+		return nil, errors.Wrapf(err, "unable to update Deployment %s", deploy.Name)
 	}
 	return deployment, nil
 }
 
 // UpdateDeployment updates a deployment based on the given deployment spec
 func (c *Client) UpdateDeployment(deploy appsv1.Deployment) (*appsv1.Deployment, error) {
-	deployment, err := c.KubeClient.AppsV1().Deployments(c.Namespace).Update(&deploy)
+	deployment, err := c.KubeClient.AppsV1().Deployments(c.Namespace).Update(context.TODO(), &deploy, metav1.UpdateOptions{FieldManager: "odo"})
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to update Deployment %s", deploy.Name)
+	}
+	return deployment, nil
+}
+
+// ApplyDeployment updates a deployment based on the given deployment spec
+func (c *Client) ApplyDeployment(deploy appsv1.Deployment) (*appsv1.Deployment, error) {
+	data, err := json.Marshal(deploy)
+
+	klog.V(5).Infoln("Applying Deployment via server-side apply:")
+	klog.V(5).Infoln(resourceAsJson(deploy))
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "unable to marshal deployment")
+	}
+	deployment, err := c.KubeClient.AppsV1().Deployments(c.Namespace).Patch(context.TODO(), deploy.Name, types.ApplyPatchType, data, metav1.PatchOptions{FieldManager: "odo", Force: boolPtr(true)})
```

Comment:
One could just read the docs, but can you add a comment explaining what's going on here in context of odo?

Reply:
Do you have any specific part in mind?
This is just calling ServerSide to apply. it is equivalent of `kubectl apply --server-side --force-conflicts`.
I'm using force to make sure that if someone changed one of the values that odo manages, odo overrides it with the value it expects instead of failing due to conflict.

Reply:
`Force` requires a pointer to `boolean` the `boolPtr` functions return point to the boolean with value defined as an argument.